app/views/responses/_answer_sets.html.erb

```diff
@@ -0,0 +1,12 @@
+<% items.each_with_index do |(item, subitems), lindex| %>
+  <% if item.is_a? QingGroup %>
+    <div class="form_field response_form_id">
+      <div class="label_and_control">
+        <label class="main"><%= item.group_name %></label>
+      </div>
+    </div>
+    <%= render 'answer_sets', items: subitems, answer_sets: answer_sets, level: level + 1, index: lindex %>
+  <% else %>
+    <%= render 'answer_set', answer_set: answer_sets[item], index: lindex + index, level: level %>
```

Comment:
`index` never gets set here for root-level questions as far as I can tell. I'm getting a nil error.